src/autocomplete/lib/autocomplete-parse-result.ts

```diff
@@ -0,0 +1,594 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+// either express or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+export interface WeightedKeywordSuggestion {
+    value: string;
+    weight: number;
+}
+
+export type KeywordSuggestion = WeightedKeywordSuggestion | string;
+
+export interface IdentifierChainEntry {
+    name?: string;
+    asterisk?: boolean;
+    cte?: string;
+    subQuery?: string;
+}
+
+export interface SubQuery {
+    alias: string;
+    columns: ColumnDetails[];
+    subQueries?: SubQuery[];
+}
+
+export interface ParsedTable {
+    alias?: string;
+    identifierChain?: IdentifierChainEntry[];
+    owner?: string;
+    subQuery?: string;
+    subQueryAlias?: string;
+}
+
+export interface ParsedLocation {
+    first_line: number;
+    first_column: number;
+    last_line: number;
+    last_column: number;
+}
+
+type IdentifierLocationType =
+    | 'alias'
+    | 'asterisk'
+    | 'column'
+    | 'database'
+    | 'file'
+    | 'function'
+    | 'functionArgument'
+    | 'statement'
+    | 'statementType'
+    | 'table'
+    | 'unknown'
+    | 'variable'
+    | 'subQuery'
+    | 'complex'
+    | 'selectList'
+    | 'whereClause'
+    | 'limitClause';
+
+export interface IdentifierLocation {
+    identifier?: string;
+    type: IdentifierLocationType;
+    alias?: string;
+    source?: string;
+    location: ParsedLocation;
+    function?: string;
+    missing?: boolean;
+    value?: string;
+    active?: boolean;
+    tables?: ParsedTable[];
+    colRef?: DetailedColumnReference;
+    argumentPosition?: number;
+    identifierChain?: IdentifierChainEntry[];
+    expression?: AwaitedTokenExpression;
+    parentLocation?: ParsedLocation;
+    path?: string;
+    qualified?: boolean;
+    columns?: ColumnDetails[];
+    subquery?: boolean;
+    suffix?: string;
+    linked?: boolean;
+    firstInChain?: boolean;
+    target?: string;
+}
+
+export interface DetailedColumnReference {
+    identifierChain: IdentifierChainEntry[];
+    tables?: ParsedTable[];
+    linked?: boolean;
+    owner?: string;
+}
+
+export interface ColumnAliasDetails {
+    name: string;
+    udfRef?: string;
+    types: string[];
+}
+
+export interface ColumnDetails {
+    type?: string | string[];
+    identifierChain?: IdentifierChainEntry[];
+    location?: ParsedLocation;
+    alias?: string;
+    subQuery?: string;
+    udfRef?: string;
+}
+
+export interface CommonPopularSuggestion {
+    tables?: ParsedTable[];
+    tablePrimaries?: ParsedTable[];
+    prefix?: string;
+    linked?: boolean;
+}
```

Comment:
Why do we need this interface, and especially export it? We can just define OrderBysSuggestion instead, and make GroupBysSuggestion it's alias.

Or we need a better naming